draft-dcook-ppm-dap-interop-test-design.md

```diff
@@ -337,29 +337,61 @@ requests (see {{collect-poll}}).
 
 ### `/internal/test/collect_poll` {#collect-poll}
 
-Upon receiving this command, the collector will poll the leader’s collect URL
-for the collect job associated with the provided handle, and provide the status
-and result to the test runner.
+The test runner sends this command to a collector to poll for completion of the
+collect job associated with the provided handle. The collector provides the
+status and (if available) result to the test runner.
 
 |Key|Value|
 |`handle`|The handle for a collect request from a previous invocation of `/internal/test/collect_start`. (see {{collect-start}})|
 {: title="Request JSON object structure"}
 
 |Key|Value|
-|`status`|Either `"complete"` if the result was returned, `"in progress"` if the result was not yet ready, or `"error"` if an error occurred.|
+|`status`|Either `"complete"` if the result is ready, `"in progress"` if the result is not yet ready, or `"error"` if an error occurred.|
 |`error` (optional)|An optional error message, to assist in troubleshooting. This will be included in the test runner logs.|
 |`batch_id` (if the task uses fixed size queries)|The identifier of the batch that was collected, encoded with base64url.|
 |`report_count` (if complete)|A number, reflecting the count of client reports included in this aggregated result.|
-|`result` (if complete)|The result of the aggregation. If the VDAF is of type Prio3Aes128Count or Prio3Aes128Sum, this will be a string, representing an integer in base 10. If the VDAF is of type Prio3Aes128CountVec or Prio3Aes128Histogram, this will be an array of strings, each representing an integer in base 10.|
+|`result` (if complete)|The result of the aggregation. If the VDAF is of type Prio3Aes128Count or Prio3Aes128Sum, this will be a string, representing an integer in base 10. If the VDAF is of type Prio3Aes128CountVec, Prio3Aes128Histogram, or Poplar1Aes128, this will be an array of strings, each representing an integer in base 10.|
 {: title="Response JSON object structure"}
 
 
-### Heavy Hitters
+### `/internal/test/heavy_hitters_start` {#heavy-hitters-start}
```

Comment:
Do the `heavy_hitters_start` & `heavy_hitters_poll` endpoints need to exist separately from the other endpoints, or could they be implemented as part of the test runner via the other endpoints? I think it's the latter; if so, I would recommend removing these endpoints & having the functionality be implemented as part of appropriate test runners in order to keep test API implementations simple.

Reply:
I think that's right, heavy hitters doesn't add any protocol complexity beyond providing examples of multiple aggregation parameters. For any collectors that do implement a heavy hitters functionality, they can test it on their own by mocking out responses from the leader; the rest of the DAP protocol run is irrelevant.